generators/chipyard/src/main/scala/upf/chisel/API.scala

```diff
@@ -0,0 +1,13 @@
+// See LICENSE for license details
+package chipyard.upf.chisel
+
+import firrtl.annotations.{Target, Annotation}
+import chipyard.upf.firrtl.NameUPFAnnotation
+
+abstract class ChiselNameUPFElement(scope: Target, name: String) {
+    def getAnnotations(): Seq[Annotation] = {
+        return Seq(NameUPFAnnotation(scope, name))
```

Comment:
What happens if in the `getAnnotations` function you return an empty `Seq`? Will things still error out?

Reply:
Yup still errors out, I've instead made changes to limit the amount of info needed to create the annotations (ex. no parameters for NameUPFAnnotation).